### ISSUE TITLE
Load cpu 100% macOS

### ISSUE THREAD

**Issue by klerick** [googleapis/nodejs-pubsub#445]
**Environment details**
 - OS: MacOs Mojave 
 - Node.js version: 8.12.0 and 10
 - npm version: 10.5.0 and 6.1.0
 - `@google-cloud/pubsub` version: 0.22.0 and 0.23.0

I am having a problem with a pubsub sdk.
I've tried to reproduced it on other pc and have got the same result.
If I try to run a subscribe script, I will get cpu loading on 100%
If i use linux, it will work ok.
```
const { PubSub } = require('@google-cloud/pubsub');
const pubsub = new PubSub({
  credentials: {}
});
const subscription = pubsub.subscription(topicName);
subscription.on(`message`, message => {
  console.log(`Received message ${message.id}:`);
  console.log(`\tData: ${message.data}`);
  console.log(`\tAttributes: ${message.attributes}`);
  messageCount += 1;
  message.ack();
});
```

![Screen load cpu](https://i.gyazo.com/04eb5d2b13021e43683e35786ec46430.png)

**Comment by callmehiphop** [comment]
Possible duplicate of #288

**Comment by callmehiphop** [comment]
Hi @klerick I'm having a bit of trouble reproducing this, could you provide me with a little bit more info?

* Are you using `GCLOUD_PROJECT` env var?
* Are you using default application credentials?
* Roughly how many messages are pending before creating a subscription?

**Comment by klerick** [comment]
I've tried to use GOOGLE_APPLICATION_CREDENTIALS env var, and everything works fine. But if I create PubSub object using params. A, then I get load 100% cpu and I do not get any error message.

**Comment by callmehiphop** [comment]
@klerick I just made a [release](https://github.com/googleapis/nodejs-pubsub/pull/452) (`0.24.0`) that should resolve the issue you're seeing. If you continue to encounter this issue, please let us know! Thanks again for the report!

**Comment by klerick** [comment]
@callmehiphop I've checked a new version. As I see the problem is still there

**Comment by callmehiphop** [comment]
@klerick I made a [branch](https://github.com/callmehiphop/nodejs-pubsub/tree/dg--445) with a possible fix, would you be willing to test it out for me?

**Comment by callmehiphop** [comment]
So I think I've narrowed this down to an auth issue (https://github.com/googleapis/google-auth-library-nodejs/issues/614), although I'm waiting on a response of some kind, I think the fix for this will be small and hopefully coming soon.

**Comment by callmehiphop** [comment]
@klerick good news, I think this has been fixed in the latest [google-auth release](https://github.com/googleapis/google-auth-library-nodejs/releases/tag/v3.1.0). Its a minor change so we won't be releasing a new version of PubSub, but you might need to reinstall to pick up the latest auth changes.

I'm going to close this (again), but if your issue persists please let me know! Thanks again!
